inst/rmd/h/default.html

```diff
@@ -677,7 +677,7 @@ <h2 id="$idprefix$toc-title">$toc-title$</h2>
       hashGenerator: function (text) {
         return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
       },
-      ignoreSelector: ".toc-ignore",
```

Comment:
I believe you added `.unlisted.unnumbered *` to handle the headers when classes is on the section divs. 

This is done for `toc-ignore ` by moving the class the header. See above (https://github.com/rstudio/rmarkdown/pull/1993/files#diff-91b59569aab1b937e1e5a6e1559a62de19e04f4c4bad52bfb08b910281e0f2dfL667) 

I don't know if that cause an issue to ignore everything under a div of class - I don't think so but this is not currently how it works with `toc-ignore` so I am wondering. I think it would be better to use the same trick no matter the classes. Easier to maintain. 

Also, BTW, the classes are not on the header because we remove it in `header-attrs.js` to insure compatibility with Pandoc<2.8 - at the end if seems odd in this kind of case but understandable to keep a similar HTML structure to manipulate no matter Pandoc version.

Reply:
Sorry, `.unlisted.unnumbered *` comes from my misunderstanding. I confirmed things work well even if removing that selector.

Reply:
Oh I was not sure it would work... 🤔 
Why do we need to move `.toc-ignore` to headers then ? 

I'll check something later

Reply:
Oops, sorry. It's my mistake...
I was using `.toc-ignore` instead of `.unlisted .unnumbered`.
Yes, we need to move `.toc-ignore` to headers.

Reply:
Now I modified the JavaScript to add `.toc-inore` selector to headers if they are children of `div.unlisted.unnumbered`.